### PR TITLE
Update reanimated to v3

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

Right now running Expo Go causes this issue:

image

A fix for this is to upgrade vendored reanimated to V3.

# How

``` et update-vendored-module -m react-native-reanimated -c 3.0.2```

CC @Kudo 
A word of warning:

We saw some issues during the linking process on Android with this new version.

```
ld: error: undefined symbol: facebook::hermes::HermesRuntime::enableSamplingProfiler()
>>> referenced by HermesSamplingProfiler.cpp:17 (/Users/aleksander/Work/Expo/expo/react-native-lab/react-native/ReactAndroid/src/main/jni/react/hermes/instrumentation/HermesSamplingProfiler.cpp:17)
```

We believe this is caused by missing .so files in 
```
react-native-lab/react-native/ReactAndroid/build/intermediates/cxx/Debug/******/obj/arm64-v8a
```
In particular, 
```
libhermes_executor.so
```
is sometimes missing during the build process.

A current fix for this was to deep clean the repo and caches, which seemed to fix the issue, or build hermes manually from gradle. Reanimated v3 uses prefabs which can cause the issue, so we might need to add some dependsOn to gradle if this occurs for you during building home.

# Test Plan

Tested using native-component-list in Expo Go on both platforms.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/vendoring/config/expoGoConfig.ts [hunk 1]
```diff
@@ -167,38 +167,38 @@ const config: VendoringTargetConfig = {
...
+            //   paths: 'CMakeLists.txt',
+            //   find: /(string\(APPEND CMAKE_CXX_FLAGS " -DJS_RUNTIME_HERMES=1"\))/g,
+            //   replaceWith: `find_package(hermes-engine REQUIRED CONFIG)\n    $1`,
+            // },
+            // {
+            //   // find hermes from prefab
+            //   paths: 'CMakeLists.txt',
+            //   find: /"\$\{BUILD_DIR\}\/.+\/libhermes\.so"/g,
+            //   replaceWith: `hermes-engine::libhermes`,
+            // },
```
Comment (Kudo) [inline tools/src/vendoring/config/expoGoConfig.ts:201]: thanks for commenting out these. i think these code are not necessary now. let's just remove them.

File: android/vendored/unversioned/react-native-reanimated/Common/cpp/ReanimatedRuntime/RuntimeInitialization.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
+# Hermes Runtime initialization
```
Comment (Kudo) [inline android/vendored/unversioned/react-native-reanimated/Common/cpp/ReanimatedRuntime/RuntimeInitialization.md:1]: would be good to exclude *.md from vendoring.

**Review by Kudo — COMMENTED** [review]
let's check the strange build error after merging.

and before that, i would like to double confirm whether layout animation is still work, because we had a different setup for layout animation.

**Comment by Kudo** [comment]
there's a [ci failure](https://github.com/expo/expo/actions/runs/4930509159/jobs/8811507692?pr=22218) in
```
/Users/runner/work/expo/expo/ios/Exponent/Versioned/Core/EXVersionManager.mm:47:9: fatal error: 'RNReanimated/REAModule.h' file not found
#import <RNReanimated/REAModule.h>
```

is it related?

**Comment by aleqsio** [comment]
@Kudo 
I've confirmed this builds on iOS with layout animations working. Also removed vendoring of .md files from reanimated.

Could still use your help with figuring out how to fix the android build flake :/

**Review by Kudo — APPROVED** [review]
checked with test suite ci and it's green: https://github.com/expo/expo/actions/runs/4981857364
let's ship it. thanks for the great upgrade!

**Review by tsapeta — APPROVED** [review]

**Pull request merged**